xarray/core/dataset.py

```diff
@@ -4284,7 +4284,7 @@ def reduce(
 
     def map(
         self,
-        func: Callable,
+        func: Union[Callable, str],
```

Comment:
I'm not opposed to this; though is there an advantage relative to passing the function directly?

Reply:
Please see my separate comment on the subject.

Reply:
I'm not a big fan of this as currently written. The main reason is consistency: we don't support strings instead of functions in other xarray methods, like `Dataset.reduce`.

We could consider this more broadly, but I would rather decouple this from this change.

Reply:
This has been removed.